motionplan/utils.go

```diff
@@ -114,3 +115,30 @@ func stepsToNodes(steps [][]referenceframe.Input) []node {
 	}
 	return nodes
 }
+
+type resultPromise struct {
+	steps  [][]referenceframe.Input
+	future chan *rrtPlanReturn
```

Comment:
what is `future` trying to convey here?

Reply:
`future` refers to this chan having a result at some point in the future. The `resultPromise` either has the result from instantiation, or else it will at some point in the future, via this channel.